src/network.zig

```diff
@@ -1203,44 +1180,482 @@ pub const Protocols = struct {
 			writer.writeEnum(items.Inventory.Command.PayloadType, payloadType);
 			std.debug.assert(writer.data.items[0] != 0xff);
 			writer.writeSlice(_data);
-			conn.sendImportant(id, writer.data.items);
+			conn.send(.fast, id, writer.data.items);
 		}
 		pub fn sendConfirmation(conn: *Connection, _data: []const u8) void {
 			std.debug.assert(conn.user != null);
 			var writer = utils.BinaryWriter.initCapacity(main.stackAllocator, _data.len + 1);
 			defer writer.deinit();
 			writer.writeInt(u8, 0xff);
 			writer.writeSlice(_data);
-			conn.sendImportant(id, writer.data.items);
+			conn.send(.fast, id, writer.data.items);
 		}
 		pub fn sendFailure(conn: *Connection) void {
 			std.debug.assert(conn.user != null);
-			conn.sendImportant(id, &.{0xfe});
+			conn.send(.fast, id, &.{0xfe});
 		}
 		pub fn sendSyncOperation(conn: *Connection, _data: []const u8) void {
 			std.debug.assert(conn.user != null);
 			var writer = utils.BinaryWriter.initCapacity(main.stackAllocator, _data.len + 1);
 			defer writer.deinit();
 			writer.writeInt(u8, 0);
 			writer.writeSlice(_data);
-			conn.sendImportant(id, writer.data.items);
+			conn.send(.fast, id, writer.data.items);
 		}
 	};
 };
 
 pub const Connection = struct { // MARK: Connection
-	const maxPacketSize: u32 = 65507; // max udp packet size
+	const maxMtu: u32 = 65507; // max udp packet size
 	const importantHeaderSize: u32 = 5;
-	const maxImportantPacketSize: u32 = 576 - 20 - 8; // IPv4 MTU minus IP header minus udp header
+	const minMtu: u32 = 576 - 20 - 8; // IPv4 MTU minus IP header minus udp header
 	const headerOverhead = 20 + 8 + 42; // IP Header + UDP Header + Ethernet header/footer
 	const congestionControl_historySize = 16;
 	const congestionControl_historyMask = congestionControl_historySize - 1;
 	const minimumBandWidth = 10_000;
-	const timeUnit = 100_000_000;
+
+	const receiveBufferSize = 8 << 20;
 
 	// Statistics:
 	pub var packetsSent: Atomic(u32) = .init(0);
 	pub var packetsResent: Atomic(u32) = .init(0);
+	pub var internalMessageOverhead: Atomic(usize) = .init(0);
+	pub var internalHeaderOverhead: Atomic(usize) = .init(0);
+	pub var externalHeaderOverhead: Atomic(usize) = .init(0);
+
+	const SequenceIndex = i32;
+
+	const RangeBuffer = struct { // MARK: RangeBuffer
```

Comment:
Ik that this has no use case outside `network.zig`, but this is a generic data structure not tied directly to `network.zig` so maybe we could move it to utils alongside other generic data strucutres?

Reply:
Yeah, I thought about making it generic, but this is hardcoded to work with wrapped integer arithmetic which is probably useless everywhere else.